### PR TITLE
[SPARK-48025][SQL][TESTS] Fix org.apache.spark.sql.execution.benchmark.DateTimeBenchmark

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR fixes several issues in org.apache.spark.sql.execution.benchmark.DateTimeBenchmark
- Misuse `trunc` function, a.k.a, the parameter order is wrong in reverse order
- Some benchmarks not compatible with ANSI by default

### Why are the changes needed?

restore benchmark cases

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

benchmark

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DateTimeBenchmark.scala [hunk 1]
```diff
@@ -75,7 +81,7 @@ object DateTimeBenchmark extends SqlBasedBenchmark {
             doBenchmark(N, s"$dt + interval 1 month 2 day")
           }
           benchmark.addCase("date + interval(m, d, ms)") { _ =>
-            doBenchmark(N, s"$dt + interval 1 month 2 day 5 hour")
+            doBenchmarkAnsiOff(N, s"$dt + interval 1 month 2 day 5 hour")
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DateTimeBenchmark.scala:84]: illegal hour portion for ansi date add

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DateTimeBenchmark.scala [hunk 2]
```diff
@@ -161,7 +167,7 @@ object DateTimeBenchmark extends SqlBasedBenchmark {
           }
           val dateExpr = "cast(timestamp_seconds(id) as date)"
           Seq("year", "yyyy", "yy", "mon", "month", "mm").foreach { level =>
-            run(N, s"trunc $level", s"trunc('$level', $dateExpr)")
+            run(N, s"trunc $level", s"trunc($dateExpr, '$level')")
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DateTimeBenchmark.scala:170]: the parameter order is wrong

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DateTimeBenchmark.scala [hunk 3]
```diff
@@ -171,7 +177,7 @@ object DateTimeBenchmark extends SqlBasedBenchmark {
           run(n, "to timestamp str", timestampStrExpr)
           run(n, "to_timestamp", s"to_timestamp($timestampStrExpr, $pattern)")
           run(n, "to_unix_timestamp", s"to_unix_timestamp($timestampStrExpr, $pattern)")
-          val dateStrExpr = "concat('2019-01-', lpad(mod(id, 25), 2, '0'))"
+          val dateStrExpr = "concat('2019-01-', lpad(mod(id, 25) + 1, 2, '0'))"
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DateTimeBenchmark.scala:180]: avoid invalid date value `2019-01-00`

File: sql/core/benchmarks/DateTimeBenchmark-jdk21-results.txt [hunk 4]
```diff
@@ -2,460 +2,460 @@
...
 
-OpenJDK 64-Bit Server VM 21.0.2+13-LTS on Linux 6.5.0-1016-azure
+OpenJDK 64-Bit Server VM 21.0.3+9-LTS on Linux 6.5.0-1018-azure
 AMD EPYC 7763 64-Core Processor
 trunc mm:                                 Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-trunc mm wholestage off                             297            298           1         33.6          29.7       1.0X
-trunc mm wholestage on                              274            276           2         36.5          27.4       1.1X
+trunc mm wholestage off                             846            847           1         11.8          84.6       1.0X
+trunc mm wholestage on                              834            845          11         12.0          83.4       1.0X
```
Comment (yaooqinn) [inline sql/core/benchmarks/DateTimeBenchmark-jdk21-results.txt:396]: The results are corrected

**Review by HyukjinKwon — APPROVED** [review]
Seems fine but do we have this issue in other banchmarks too?

**Comment by yaooqinn** [comment]
Thank you @HyukjinKwon. I have dispatched a CI task to audit, https://github.com/yaooqinn/spark/actions/runs/8866497258

**Comment by yaooqinn** [comment]
Merged to master

**Pull request closed without merging**